pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go

```diff
@@ -0,0 +1,926 @@
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// *Optional* +
+	//
+	// Specify node affinity, pod affinity, and pod anti-affinity for LogSearch API pods. +
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// *Optional* +
+	//
+	// Specify one or more https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/[Kubernetes tolerations] to apply to MinIO Log Search API pods.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// *Optional* +
+	//
+	// If provided, use these annotations for Log Search Object Meta annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// *Optional* +
+	//
+	// If provided, use these labels for Log Search Object Meta labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// *Optional* +
+	//
+	// Object specification for configuring the backing PostgreSQL database for the LogSearch API. +
+	// +optional
+	Db *LogDbConfig `json:"db,omitempty"` // nolint:stylecheck
+	// *Required* +
+	//
+	// Object specification for configuring LogSearch API.
+	// +optional
+	Audit *AuditConfig `json:"audit,omitempty"`
+	// *Optional* +
+	//
+	// Specify the https://kubernetes.io/docs/tasks/configure-pod-container/security-context/[Security Context] of pods deployed as part of the Log Search API. The Operator supports only the following pod security fields: +
+	//
+	// * `fsGroup` +
+	//
+	// * `fsGroupChangePolicy` +
+	//
+	// * `runAsGroup` +
+	//
+	// * `runAsNonRoot` +
+	//
+	// * `runAsUser` +
+	//
+	// * `seLinuxOptions` +
+	// +optional
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+}
+
+// AuditConfig defines configuration parameters for Audit (type) logs.
+type AuditConfig struct {
+	// *Required* +
+	//
+	// Specify the amount of storage to request in Gigabytes (GB) for storing audit logs.
+	// +optional
+	DiskCapacityGB *int `json:"diskCapacityGB,omitempty"`
+}
+
+// PrometheusConfig (`prometheus`) defines the configuration of a Prometheus instance as part of the MinIO tenant. The Operator automatically configures the Prometheus instance to scrape and store metrics from the MinIO tenant. +
+//
+// The Operator deploys each Prometheus pod using the {prometheus-image} Docker image.
+type PrometheusConfig struct {
+	// *Optional* +
+	//
+	// Defines the Docker image to use for deploying Prometheus pods. Defaults to {prometheus-image}. +
+	// +optional
+	Image string `json:"image,omitempty"`
+	// *Optional* +
+	//
+	// Deprecated: in Operator v4.0.1* +
+	//
+	// Defines the Docker image to use as a sidecar for the Prometheus server. Defaults to `alpine`. +
+	//
+	// The specified Docker image *must* be the https://hub.docker.com/_/alpine[`alpine`] package. +
+	// +optional
+	SideCarImage string `json:"sidecarimage,omitempty"`
```

Comment:
Not sure why this structure is under `.../storage/minio/.../types.go`. I suggest to split this `types.go` file to multiple smaller one.
WDYT?